### PR TITLE
Setup addon tests

### PR DESCRIPTION
**Pull request by xg-wang**
Introduce [ember-cli-addon-tests](https://github.com/tomdale/ember-cli-addon-tests) to test:

1. ember app with `ember-cli-fastboot` has `dist/ember-fetch/fastboot-fetch.js` built
2. serve ember app with `ember-cli-fastboot` renders fetched content

### CODE REVIEW

**Comment by xg-wang** [comment]
Blocked by https://github.com/tomdale/ember-cli-addon-tests/issues/198 and https://github.com/tomdale/ember-cli-addon-tests/issues/199

**Comment by xg-wang** [comment]
@Turbo87 @stefanpenner @rwjblue This is green now, can you review?

File: test/fixtures/dummy/app/app.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+const App = Application.extend({
+  modulePrefix: config.modulePrefix,
+  podModulePrefix: config.podModulePrefix,
+  Resolver
+});
+
+loadInitializers(App, config.modulePrefix);
+
+export default App;
```
Comment (Turbo87) [inline test/fixtures/dummy/app/app.js]: where are these `fixture` files being used?

File: test/fastboot-test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    app = new AddonTestApp();
+
+    return app
+      .create('dummy', { skipNpm: true })
+      .then(app =>
+        app.editPackageJSON(pkg => {
+          pkg.devDependencies['ember-cli-fastboot'] = '*';
+          // These 2 are in ember-fetch's package.json, symlinking to dummy won't help resolve
+          pkg.devDependencies['abortcontroller-polyfill'] = '*';
+          pkg.devDependencies['node-fetch'] = '*';
```
[the remark this answers is not in the document]
Comment (xg-wang) [reply test/fastboot-test.js]: See this [PR](https://github.com/ember-fastboot/fastboot/pull/200), `Fastboot.requrie` is reading from `node_modules/node-fetch` but it's under `node_modules/ember-fetch/node-fetch` when symlink.

File: test/fixtures/dummy/app/app.js (hunk 1 above)
Comment (xg-wang) [reply test/fixtures/dummy/app/app.js]: These are used by `ember-cli-addon-tests` to copy over generated new app. It's same setup as https://github.com/ember-fastboot/ember-cli-fastboot/tree/master/test

File: .travis.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ jobs:
       script:
         - yarn lint:js
         - yarn test
-        - yarn test:node
+        - npm run test:node
```
Comment (tomdale) [inline .travis.yml]: Why change this to npm?

Comment (xg-wang) [reply .travis.yml]: This is a known issue https://github.com/tomdale/ember-cli-addon-tests/issues/198, should be fixed by a WIP PR https://github.com/tomdale/ember-cli-addon-tests/pull/105

**Pull request merged**